test/components/views/right_panel/RoomSummaryCard-test.tsx

```diff
@@ -0,0 +1,182 @@
+import { MatrixEvent, Room } from "matrix-js-sdk/src/matrix";
+
+import DMRoomMap from "../../../../src/utils/DMRoomMap";
+import RoomSummaryCard from "../../../../src/components/views/right_panel/RoomSummaryCard";
+import ShareDialog from "../../../../src/components/views/dialogs/ShareDialog";
+import ExportDialog from "../../../../src/components/views/dialogs/ExportDialog";
+import MatrixClientContext from "../../../../src/contexts/MatrixClientContext";
+import defaultDispatcher from "../../../../src/dispatcher/dispatcher";
+import * as settingsHooks from "../../../../src/hooks/useSettings";
+import Modal from "../../../../src/Modal";
+import RightPanelStore from "../../../../src/stores/right-panel/RightPanelStore";
+import { RightPanelPhases } from "../../../../src/stores/right-panel/RightPanelStorePhases";
+import { getMockClientWithEventEmitter, mockClientMethodsUser } from "../../../test-utils";
+import { PollHistoryDialog } from "../../../../src/components/views/dialogs/polls/PollHistoryDialog";
+
+describe("<RoomSummaryCard />", () => {
+    const userId = "@alice:domain.org";
+    const mockClient = getMockClientWithEventEmitter({
+        ...mockClientMethodsUser(userId),
+        isRoomEncrypted: jest.fn(),
+        getRoom: jest.fn(),
+    });
+    const roomId = "!room:domain.org";
+    const room = new Room(roomId, mockClient, userId);
+    const roomCreateEvent = new MatrixEvent({
+        type: "m.room.create",
+        room_id: roomId,
+        sender: userId,
+        content: {
+            creator: userId,
+            room_version: "5",
+        },
+        state_key: "",
+    });
+    room.currentState.setStateEvents([roomCreateEvent]);
+    const defaultProps = {
+        room,
+        onClose: jest.fn(),
+    };
+    const getComponent = (props = {}) =>
+        render(<RoomSummaryCard {...defaultProps} {...props} />, {
+            wrapper: ({ children }) => (
+                <MatrixClientContext.Provider value={mockClient}>{children}</MatrixClientContext.Provider>
+            ),
+        });
+
+    const modalSpy = jest.spyOn(Modal, "createDialog");
+    const dispatchSpy = jest.spyOn(defaultDispatcher, "dispatch");
+    const rightPanelCardSpy = jest.spyOn(RightPanelStore.instance, "pushCard");
+    const featureEnabledSpy = jest.spyOn(settingsHooks, "useFeatureEnabled");
+
+    beforeEach(() => {
+        jest.clearAllMocks();
+        DMRoomMap.makeShared();
+
+        mockClient.getRoom.mockReturnValue(room);
+        jest.spyOn(room, "isElementVideoRoom").mockRestore();
+        jest.spyOn(room, "isCallRoom").mockRestore();
+        featureEnabledSpy.mockReset().mockReturnValue(false);
+    });
+
+    it("renders the room summary", () => {
+        const { container } = getComponent();
+        expect(container).toMatchSnapshot();
+    });
+
+    it("opens room members list on button click", () => {
+        const { getByTestId } = getComponent();
+
+        fireEvent.click(getByTestId("roomMembersButton"));
```

Comment:
We should use `getByText` or `getByLabelText` here. Because users don't find elements by test-id.